File: back/src/main/java/com/triptrace/domain/trip/trip/entity/Trip.java
package com.triptrace.domain.trip.trip.entity;

import com.triptrace.domain.image.image.entity.Image;
import com.triptrace.domain.member.member.entity.Member;
import com.triptrace.global.jpa.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import java.time.LocalDateTime;

@Entity
public class Trip extends BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private Member owner;

    @Column(nullable = false)
    private String title;

    private String country;

    private String city;

    private LocalDateTime startDate;

    private LocalDateTime endDate;

    @Column(nullable = false)
    private boolean visibility;

    @Column(nullable = false)
    private Long likeCount = 0L;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "representative_image_id")
    private Image representativeImage;

    public void increaseLikeCount() {
        this.likeCount++;
    }

    public void decreaseLikeCount() {
        this.likeCount--;
    }

    public void modify(
        String title,
        String country,
        String city,
        LocalDateTime startDate,
        LocalDateTime endDate,
        boolean visibility
    ) {
        this.title = title;
        this.country = country;
        this.city = city;
        this.startDate = startDate;
        this.endDate = endDate;
        this.visibility = visibility;
    }

    public void changeRepresentativeImage(Image representativeImage) {
        this.representativeImage = representativeImage;
    }

    public void changeAutoRecordDefaults(
        String country,
        String city,
        LocalDateTime startDate,
        LocalDateTime endDate
    ) {
        this.country = country;
        this.city = city;
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public void changeDateRange(LocalDateTime startDate, LocalDateTime endDate) {
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public Trip(
        Member owner,
        String title,
        String country,
        String city,
        LocalDateTime startDate,
        LocalDateTime endDate,
        boolean visibility
    ) {
        this.owner = owner;
        this.title = title;
        this.country = country;
        this.city = city;
        this.startDate = startDate;
        this.endDate = endDate;
        this.visibility = visibility;
    }

    @java.lang.SuppressWarnings("all")
    public Member getOwner() {
        return this.owner;
    }

    @java.lang.SuppressWarnings("all")
    public String getTitle() {
        return this.title;
    }

    @java.lang.SuppressWarnings("all")
    public String getCountry() {
        return this.country;
    }

    @java.lang.SuppressWarnings("all")
    public String getCity() {
        return this.city;
    }

    @java.lang.SuppressWarnings("all")
    public LocalDateTime getStartDate() {
        return this.startDate;
    }

    @java.lang.SuppressWarnings("all")
    public LocalDateTime getEndDate() {
        return this.endDate;
    }

    @java.lang.SuppressWarnings("all")
    public boolean isVisibility() {
        return this.visibility;
    }

    @java.lang.SuppressWarnings("all")
    public Long getLikeCount() {
        return this.likeCount;
    }

    @java.lang.SuppressWarnings("all")
    public Image getRepresentativeImage() {
        return this.representativeImage;
    }

    @java.lang.SuppressWarnings("all")
    public Trip() {
    }
}

File: back/src/main/java/com/triptrace/domain/trip/trip/entity/Trip.kt
package com.triptrace.domain.trip.trip.entity

import com.triptrace.domain.image.image.entity.Image
import com.triptrace.domain.member.member.entity.Member
import com.triptrace.global.jpa.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalDateTime

@Entity
class Trip : BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    lateinit var owner: Member
        protected set

    @Column(nullable = false)
    lateinit var title: String
        protected set

    var country: String? = null
        protected set

    var city: String? = null
        protected set

    var startDate: LocalDateTime? = null
        protected set

    var endDate: LocalDateTime? = null
        protected set

    @Column(nullable = false)
    private var visibility: Boolean = false

    @Column(nullable = false)
    var likeCount: Long = 0L
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "representative_image_id")
    var representativeImage: Image? = null
        protected set

    fun increaseLikeCount() {
        likeCount++
    }

    fun decreaseLikeCount() {
        likeCount--
    }

    fun isVisibility(): Boolean = visibility

    fun modify(
        title: String,
        country: String?,
        city: String?,
        startDate: LocalDateTime?,
        endDate: LocalDateTime?,
        visibility: Boolean
    ) {
        this.title = title
        this.country = country
        this.city = city
        this.startDate = startDate
        this.endDate = endDate
        this.visibility = visibility
    }

    fun changeRepresentativeImage(representativeImage: Image?) {
        this.representativeImage = representativeImage
    }

    fun changeAutoRecordDefaults(
        country: String?,
        city: String?,
        startDate: LocalDateTime?,
        endDate: LocalDateTime?
    ) {
        this.country = country
        this.city = city
        this.startDate = startDate
        this.endDate = endDate
    }

    fun changeDateRange(startDate: LocalDateTime?, endDate: LocalDateTime?) {
        this.startDate = startDate
        this.endDate = endDate
    }

    constructor(
        owner: Member,
        title: String,
        country: String?,
        city: String?,
        startDate: LocalDateTime?,
        endDate: LocalDateTime?,
        visibility: Boolean
    ) {
        this.owner = owner
        this.title = title
        this.country = country
        this.city = city
        this.startDate = startDate
        this.endDate = endDate
        this.visibility = visibility
    }

    constructor()
}
